miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultProjectService.java

```diff
@@ -48,267 +47,180 @@
-    this.platformStore = platformStore;
-  }
-
-  public void setPoolStore(PoolStore poolStore) {
-    this.poolStore = poolStore;
-  }
-
-  public void setProjectStore(ProjectStore projectStore) {
-    this.projectStore = projectStore;
-  }
-
-  public void setReferenceGenomeStore(ReferenceGenomeDao referenceGenomeStore) {
-    this.referenceGenomeDao = referenceGenomeStore;
-  }
-
-  public void setRunStore(RunStore runStore) {
-    this.runStore = runStore;
-  }
-
-  public void setSequencerPartitionContainerStore(SequencerPartitionContainerStore sequencerPartitionContainerStore) {
-    this.sequencerPartitionContainerStore = sequencerPartitionContainerStore;
-  }
-
-  public void setSequencerReferenceStore(SequencerReferenceStore sequencerReferenceStore) {
-    this.sequencerReferenceStore = sequencerReferenceStore;
+  @Override
+  public void addProjectWatcher(Project project, User watcher) throws IOException {
+    authorizationManager.throwIfNotWritable(project);
+    projectStore.addWatcher(project, watcher);
   }
 
-  public void setSubmissionStore(SubmissionStore submissionStore) {
-    this.submissionStore = submissionStore;
+  @Override
+  public void deleteProjectOverviewNote(ProjectOverview projectOverview, Long noteId) throws IOException {
+    authorizationManager.throwIfNonAdmin();
+    if (noteId == null || noteId.equals(Note.UNSAVED_ID)) {
+      throw new IllegalArgumentException("Cannot delete an unsaved Note");
+    }
+    ProjectOverview managed = projectStore.getProjectOverviewById(projectOverview.getId());
+    Note deleteNote = null;
+    for (Note note : managed.getNotes()) {
+      if (note.getNoteId().equals(noteId)) {
+        deleteNote = note;
+        break;
+      }
+    }
+    if (deleteNote == null) {
+      throw new IOException("Note " + noteId + " not found for ProjectOverview " + projectOverview.getId());
+    }
+    managed.getNotes().remove(deleteNote);
+    projectStore.saveOverview(managed);
   }
 
-  public void setAutoGenerateIdBarcodes(boolean autoGenerateIdBarcodes) {
-    this.autoGenerateIdBarcodes = autoGenerateIdBarcodes;
+  public User getCurrentUser() throws IOException {
+    Authentication auth = SecurityContextHolder.getContextHolderStrategy().getContext().getAuthentication();
+    if (auth == null) {
+      return null;
+    }
+    User user = securityManager.getUserByLoginName(auth.getName());
+    if (user == null && auth.isAuthenticated()) {
+      user = new UserImpl();
+      user.setAdmin(true);
+      user.setActive(true);
+    }
+    return user;
   }
 
   @Override
-  public Collection<Project> listAllProjects() throws IOException {
+  public Project getProjectByAlias(String projectAlias) throws IOException {
     if (projectStore != null) {
-      return projectStore.listAll();
+      return projectStore.getByAlias(projectAlias);
     } else {
       throw new IOException("No projectStore available. Check that it has been declared in the Spring config.");
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "No projectStore available. Check that it has been declared in the Spring config." 6 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)